src/solver/optim-model-filler/ComponentFiller.cpp

```diff
@@ -54,21 +175,37 @@ void ComponentFiller::addVariables(Optimisation::LinearProblemApi::ILinearProble
         return;
     }
 
-    Expressions::Visitors::EvalVisitor evaluator(evaluationContext_);
+    Expressions::Visitors::EvaluationContext evaluationContext(component_.getParameterValues(),
+                                                               {},
+                                                               data);
+
+    // TODO Optimisation::LinearProblemApi::DataSeriesKeys
+    Expressions::Visitors::EvalVisitor evaluator(evaluationContext,
+                                                 {.fillContext = ctx,
+                                                  .scenarioGroup = component_.getScenarioGroupId(),
+                                                  .scenario = 0});
     for (const auto& variable: component_.getModel()->Variables() | std::views::values)
     {
+        const auto& lb = evaluator.dispatch(variable.LowerBound().RootNode());
+        const auto& ub = evaluator.dispatch(variable.UpperBound().RootNode());
         if (variable.isTimeDependent())
         {
-            pb.addVariable(evaluator.dispatch(variable.LowerBound().RootNode()),
-                           evaluator.dispatch(variable.UpperBound().RootNode()),
-                           variable.Type() != Study::SystemModel::ValueType::FLOAT,
-                           component_.Id() + "." + variable.Id(),
-                           ctx.getNumberOfTimestep());
+            std::visit(
```

Comment:
Please add a comment here to explain that you're using `std::visit` to handle the 4 cases: double/double, vector/double, double/vector and vector/vector.